### DISCUSSION TITLE
How to merge one branch (repo 1) to another branch (repo 2) for commits between two Hash numbers

### DISCUSSION THREAD

**Discussion by ashishsh51911** [community/community#21766] [New to GitHub]
<p>i have a scenario where i am trying to merge one branch in repo A to another branch in repo B,<br>
i was able to figure out the commands for doing so but it seems that the merge is happening for all the commits.<br>
I need to merge the changes between two Hash Number range.</p>
<p>say i want to merge Branch1 (Repo1) --&gt; Branch2 (Repo2) within Branch 1 Hash numbers range 2393878-2349293<br>
to Branch 2</p>
<p>here is my code: -</p>
<pre><code>git remote add old-project ../source-project
git fetch source-project
git merge -S --allow-unrelated-histories source-project/master // on this line can i put a hash range to merge ?
git push destination-project feature/source-project
git remote rm source-project
</code></pre>
<p>I am new to GIT so requesting brief descriptions along with your answers. Also if there is a better code approach i would request you to post that.<br>
As with current code, the merge conflicts are coming something like this below (the whole file is conflicted even though it should not be) :- <br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/044652851161c8d5af3a639d2964a5a782479c67.png" data-download-href="/uploads/short-url/BOzDyxcvzMiPQVCUvBj8GtmuPR.png?dl=1" title="8TsZw.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/32767257/181066794-a2e314f5-7780-4fa8-b16a-c11d3603ccb2.png" alt="8TsZw.png" data-base62-sha1="BOzDyxcvzMiPQVCUvBj8GtmuPR" width="690" height="418" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/044652851161c8d5af3a639d2964a5a782479c67_2_690x418.png, https://aws1.discourse-cdn.com/github/optimized/1X/044652851161c8d5af3a639d2964a5a782479c67_2_1035x627.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/044652851161c8d5af3a639d2964a5a782479c67_2_1380x836.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/044652851161c8d5af3a639d2964a5a782479c67_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">8TsZw.png</span><span class="informations">1928×1168 132 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
Thank You.

**Comment by logankilpatrick** [comment, marked as answer]
<p>Hey <a class="mention" href="/ashishsh51911">@ashishsh51911</a>, thanks for being here. The best route to achieve what you want would actually be to undo or alter the secondary branch such that it contains only the commits you want to be merged in as there is no good way with git to only merge certain commits between a specific range. </p>
<p>Let me know if you have any follow up questions! </p>
